packages/block-library/src/image/deprecated.js

```diff
@@ -8,353 +8,535 @@ import classnames from 'classnames';
+		return (
+			<figure
+				className={ align ? `align${ align }` : null }
+				style={ figureStyle }
+			>
+				{ href ? <a href={ href }>{ image }</a> : image }
+				{ ! RichText.isEmpty( caption ) && (
+					<RichText.Content tagName="figcaption" value={ caption } />
+				) }
+			</figure>
+		);
 	},
 };
 
-const blockSupports = {
-	anchor: true,
-	color: {
-		__experimentalDuotone: 'img',
-		text: false,
-		background: false,
-	},
-	__experimentalBorder: {
-		radius: true,
-		__experimentalDefaultControls: {
-			radius: true,
+/**
+ * Deprecation for adding the `is-resized` class to the image block to fix
+ * captions on resized images.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/6496
+ */
+const v2 = {
+	attributes: {
+		url: {
+			type: 'string',
+			source: 'attribute',
+			selector: 'img',
+			attribute: 'src',
+		},
+		alt: {
+			type: 'string',
+			source: 'attribute',
+			selector: 'img',
+			attribute: 'alt',
+			default: '',
+		},
+		caption: {
+			type: 'array',
+			source: 'children',
+			selector: 'figcaption',
+		},
+		href: {
+			type: 'string',
+			source: 'attribute',
+			selector: 'a',
+			attribute: 'href',
+		},
+		id: {
+			type: 'number',
+		},
+		align: {
+			type: 'string',
+		},
+		width: {
+			type: 'number',
+		},
+		height: {
+			type: 'number',
 		},
 	},
-};
+	save( { attributes } ) {
+		const { url, alt, caption, align, href, width, height, id } =
+			attributes;
 
-const deprecated = [
-	// The following deprecation moves existing border radius styles onto the
-	// inner img element where new border block support styles must be applied.
-	// It will also add a new `.has-custom-border` class for existing blocks
-	// with border radii set. This class is required to improve caption position
-	// and styling when an image within a gallery has a custom border or
-	// rounded corners.
-	//
-	// See: https://github.com/WordPress/gutenberg/pull/31366/
-	{
-		attributes: blockAttributes,
-		supports: blockSupports,
-		save( { attributes } ) {
-			const {
-				url,
-				alt,
-				caption,
-				align,
-				href,
-				rel,
-				linkClass,
-				width,
-				height,
-				id,
-				linkTarget,
-				sizeSlug,
-				title,
-			} = attributes;
+		const image = (
+			<img
+				src={ url }
+				alt={ alt }
+				className={ id ? `wp-image-${ id }` : null }
+				width={ width }
+				height={ height }
+			/>
+		);
 
-			const newRel = ! rel ? undefined : rel;
+		return (
+			<figure className={ align ? `align${ align }` : null }>
+				{ href ? <a href={ href }>{ image }</a> : image }
+				{ ! RichText.isEmpty( caption ) && (
+					<RichText.Content tagName="figcaption" value={ caption } />
+				) }
+			</figure>
+		);
+	},
+};
 
-			const classes = classnames( {
-				[ `align${ align }` ]: align,
-				[ `size-${ sizeSlug }` ]: sizeSlug,
-				'is-resized': width || height,
-			} );
+/**
+ * Deprecation for image floats including a wrapping div.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/7721
+ */
+const v3 = {
```

Comment:
See https://github.com/WordPress/gutenberg/blob/f9e82d00668ec88a2cc355fb33e441f85393728e/packages/block-library/src/image/index.js for the definition at the time the v3 deprecation was added.